packages/flame/lib/src/components/component.dart

```diff
@@ -122,10 +122,14 @@ class Component with Loadable {
   /// your state considering this.
   /// All components in the tree are always updated by the same amount. The time
   /// each one takes to update adds up to the next update cycle.
-  @mustCallSuper
-  void update(double dt) {
+  void update(double dt) {}
+
+  void updateTree(double dt, {bool callUpdate = true}) {
     children.updateComponentList();
-    children.forEach((c) => c.update(dt));
+    if (callUpdate) {
+      update(dt);
+    }
+    children.forEach((c) => c.updateTree(dt));
```

Comment:
The question is whether the current component should be updated before or after the children components.

Reply:
Before I would say, since it is more likely that the children depend on the state of their parent than the other way around..